spec/integration_spec.rb

```diff
@@ -345,12 +347,10 @@ def rollback
 
   context "when an around hook errors early" do
     let(:organizer) {
-      build_organizer(organize: [organizer2, interactor3, organizer4, interactor5]) do
-        around do |interactor|
+      interactors = [organizer2, interactor3, organizer4, interactor5]
+      build_organizer(organize: interactors) do
+        around do |_interactor|
           raise "foo"
-          context.steps << :around_before
-          interactor.call
-          context.steps << :around_after
```

Comment:
I like your use of `unexpected_error!` although we might have to define that method within the interactor's definition in the spec rather than as a helper method in the spec, which muddies the waters a little bit.